site2/docs/schema-manage.md

```diff
@@ -0,0 +1,497 @@
+
+#### Java Admin API
+
+To delete all versions of a schema, you can use the following method.
+
+```text
+void deleteSchema(String topic)
+```
+
+Here is an example of deleting a schema.
+
+```text
+PulsarAdmin admin = …;
+
+admin.deleteSchema("my-tenant/my-ns/my-topic"); 
+```
+
+## Custom schema storage
+
+By default, Pulsar stores various data types of schemas in [Apache BookKeeper](https://bookkeeper.apache.org) deployed alongside Pulsar. However, you can use another storage system if needed. 
+
+### Implement
+
+To use a non-default (non-BookKeeper) storage system for Pulsar schemas, you need to implement two Java interfaces: 
+
+* [`SchemaStorage`](#schemastorage-interface) 
+
+* [`SchemaStorageFactory`](#schemastoragefactory-interface)
+
+#### SchemaStorage interface
+
+The `SchemaStorage` interface has the following methods:
+
+```text
+public interface SchemaStorage {
+    // How schemas are updated
+    CompletableFuture<SchemaVersion> put(String key, byte[] value, byte[] hash);
+
+    // How schemas are fetched from storage
+    CompletableFuture<StoredSchema> get(String key, SchemaVersion version);
+
+    // How schemas are deleted
+    CompletableFuture<SchemaVersion> delete(String key);
+
+    // Utility method for converting a schema version byte array to a SchemaVersion object
+    SchemaVersion versionFromBytes(byte[] version);
+
+    // Startup behavior for the schema storage client
+    void start() throws Exception;
+
+    // Shutdown behavior for the schema storage client
+    void close() throws Exception;
+}
+```
+
+> #### Tip
+> 
+> For a complete example of **schema storage** implementation, see [BookKeeperSchemaStorage](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/BookkeeperSchemaStorage.java) class.
+
+#### SchemaStorageFactory interface 
+
+The `SchemaStorageFactory` interface has the following method:
+
+```text
+public interface SchemaStorageFactory {
+    @NotNull
+    SchemaStorage create(PulsarService pulsar) throws Exception;
+}
+```
+
+> Tip
+> 
+> For a complete example of **schema storage factory** implementation, see [BookKeeperSchemaStorageFactory](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/BookkeeperSchemaStorageFactory.java) class.
+
+### Deploy
+
+To use your custom schema storage implementation, you can perform the following steps.
+
+1. Package the implementation in a [JAR](https://docs.oracle.com/javase/tutorial/deployment/jar/basicsindex.html) file.
+   
+2. Add the JAR file to the `lib` folder in your Pulsar binary or source distribution.
+   
+3. Change the `schemaRegistryStorageClassName` configuration in `broker.conf` to your custom factory class.
+
+    For example, change the `schemaRegistryStorageClassName` to the `SchemaStorageFactory` implementation rather than the `SchemaStorage` implementation.
+      
```

Comment:
@sijie Is the sentence accurate?

Reply:
I am not sure that is correct. I think we can remove it.

Reply:
OK